runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -13255,6 +13256,452 @@ TR::Register *J9::X86::I386::TreeEvaluator::dwrtbariEvaluator(TR::Node *node, TR
+   TR::RegisterDependencyConditions  *dependencies = generateRegisterDependencyConditions((uint8_t)4, 6, cg);
+
+   dependencies->addPostCondition(lowRegister, TR::RealRegister::eax, cg);
+   dependencies->addPostCondition(highRegister, TR::RealRegister::edx, cg);
+   dependencies->addPreCondition(firstHigh, TR::RealRegister::NoReg, cg);
+   dependencies->addPreCondition(secondHigh, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(firstHigh, TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(secondHigh, TR::RealRegister::NoReg, cg);
+   dependencies->addPreCondition(firstRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+   dependencies->addPreCondition(secondRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(firstRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+   dependencies->addPostCondition(secondRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+
+   J9::IA32PrivateLinkage *linkage = static_cast<J9::IA32PrivateLinkage *>(cg->getLinkage(TR_Private));
+   TR::IA32LinkageUtils::pushLongArg(secondChild, cg);
+   TR::IA32LinkageUtils::pushLongArg(firstChild, cg);
+   TR::X86ImmSymInstruction  *instr =
+   generateHelperCallInstruction(node, TR_IA32longRemainder, dependencies, cg);
+   if (!linkage->getProperties().getCallerCleanup())
+      {
+      instr->setAdjustsFramePointerBy(-16);  // 2 long args
+      }
+
+   // Don't preserve eax and edx
+   //
+   instr->setNeedsGCMap(0xFF00FFF6);
+
+   TR::RegisterDependencyConditions  *movDependencies = generateRegisterDependencyConditions((uint8_t)6, 6, cg);
+   movDependencies->addPreCondition(lowRegister, TR::RealRegister::eax, cg);
+   movDependencies->addPreCondition(highRegister, TR::RealRegister::edx, cg);
+   movDependencies->addPostCondition(lowRegister, TR::RealRegister::eax, cg);
+   movDependencies->addPostCondition(highRegister, TR::RealRegister::edx, cg);
+   movDependencies->addPreCondition(firstHigh, TR::RealRegister::NoReg, cg);
+   movDependencies->addPreCondition(secondHigh, TR::RealRegister::NoReg, cg);
+   movDependencies->addPostCondition(firstHigh, TR::RealRegister::NoReg, cg);
+   movDependencies->addPostCondition(secondHigh, TR::RealRegister::NoReg, cg);
+   movDependencies->addPreCondition(firstRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+   movDependencies->addPreCondition(secondRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+   movDependencies->addPostCondition(firstRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+   movDependencies->addPostCondition(secondRegister->getLowOrder(), TR::RealRegister::NoReg, cg);
+
+   generateLabelInstruction(TR::InstOpCode::label, node, doneLabel, movDependencies, cg);
+
+   TR::Register *targetRegister = cg->allocateRegisterPair(lowRegister, highRegister);
+   node->setRegister(targetRegister);
+
+   return targetRegister;
+   }
+
+TR::Register *J9::X86::I386::TreeEvaluator::lstoreEvaluator(TR::Node *node, TR::CodeGenerator *cg)
```

Comment:
From what I can tell, the majority of this evaluator is the same across both OMR and OpenJ9, except for a couple of small bits of code related to `java_util_concurrent_atomic_AtomicLong_lazySet`. As such, this evaluator should just exist in OMR. See suggestions at the appropriate code locations below.